docs/core/idioms/file-based.md

```diff
@@ -0,0 +1,69 @@
+# Using file based flow storage
+
+Prefect version `0.12.1` began to implement support for storing flows as paths to files. This means that flow code can change in between (or even during) runs without needing to be reregistered. As long as the structure of the flow itself does not change, only the task content, then a Prefect API backend will be able to execute the flow. This is a useful storage mechanism especially for testing, debugging, CI/CD processes, and more!
+
+### Example file based workflow
+
+In this example we will walk through a potential workflow you may use when registering flows with [GitHub](/api/latest/environments/storage.html#github) storage. This example takes place in a GitHub repository with the following structure:
+
+```
+repo
+    README.md
+    flows/
+        my_flow.py
+```
+
+First, compose your flow file and give the flow `GitHub` storage:
+
+```python
+# flows/my_flow.py
+
+from prefect import task, Flow
+from prefect.environments.storage import GitHub
+
+@task
+def get_data():
+    return [1, 2, 3, 4, 5]
+
+@task
+def print_data(data):
+    print(data)
+
+with Flow("file-based-flow") as flow:
+    data = get_data()
+    print_data(data)
+
+flow.storage = GitHub(
+    repo="org/repo",                 # name of repo
+    path="flows/my_flow.py",        # location of flow file in repo
+    secrets=["GITHUB_ACCESS_TOKEN"]  # name of personal access token secret
+)
+```
+
+Here's a breakdown of the three kwargs set on the `GitHub` storage:
+
+- `repo`: the name of the repo that this code will live in
+- `path`: the location of the flow file in the repo. This must be an exact match to the path of the file.
+- `secrets`: the name of a [default Prefect secret](http://localhost:8081/core/concepts/secrets.html#default-secrets) which is a GitHub [personal access token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). This is set so that when the flow is executed it has the proper permissions to pull the file from the repo.
```

Comment:
```suggestion
- `secrets`: the name of a [default Prefect secret](/core/concepts/secrets.html#default-secrets) which is a GitHub [personal access token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). This is set so that when the flow is executed it has the proper permissions to pull the file from the repo.
```